### ISSUE TITLE
Impossible to reproduce experiment

### ISSUE THREAD

**Issue by smivv** [ashleve/lightning-hydra-template#168]
Hello!

Current implementation of configs saving won't allow me to reproduce experiment since:

- Currently overrides does not contain parameters so I am actually unable to override experiment params (contains only name);
- There is no written way how to load experiment configs for reproducing;

Please help with fixing these errors, they are critical!

Thanks!

**Comment by ashleve** [comment]
Hi @smivv !

After running the experiment you should be able to find `.hydra` folder in your logs:
![image](https://user-images.githubusercontent.com/43458894/131481360-4355000c-d217-4446-b16c-abd721149264.png)

`overrides.yaml` contains overridden parameters, while `config.yaml` contains full config with overrides.

You can load it to reproduce experiment:
`python run.py --config-path /logs/runs/.../.hydra --config-name config.yaml `


The only problem is that hydra output paths configuration is placed in `hydra.yaml` (and it's not present in `config.yaml`), which means that if you load `config.yaml` to reproduce the experiment, the logging folder path will revert to default. Currently I don't really see a workaround for that since this is internal Hydra behavior.


Thanks for the suggestions, I will make sure to add #Reproducibilty section to the README in the next release

**Comment by smivv** [comment]
Hi @ashleve,

Yeah, I already tested what you said and since we have overrider parameters (only their names actually), we can't reproduce everything.

E.g. I set `python run.py experiment=full`, so I will have in overrides.yaml only one line like this `- experiment=full` and that's it.
Actually the problem is that config.yaml contains placeholders like `${training.main_metric}` for example, which could be available within experiment yaml file only.

So my proposal would be to save full config after overriding all parameters and replacing placeholders in order to have fully ready config to reproduce experiment.

**Comment by ashleve** [comment]
It's true `overrides.yaml` contains only names,

The thing is, overridden parameters are also overridden in `config.yaml` from `.hydra` folder, and the configuration there should be full (meaning the names should be full config sections). Are you sure that's not the case with your project? Works exactly like that for me

Even when some experiment config contains a custom parameter like `${training.main_metric}`, for me it is still available in the `config.yaml` 
(remember that rich config printing in the terminal doesn't show every section unless it's specified in `utils.py`)

tl;dr: I still believe calling `python run.py --config-path /logs/runs/.../.hydra/ --config-name config.yaml` always reproduces the exact experiment, only hydra specific config parts (like logs output folder) should be different.
